lib/galaxy/datatypes/mothur.py

```diff
@@ -0,0 +1,937 @@
+                   axis1   axis2
+           U68589  0.262608        -0.077498
+           U68590  0.027118        0.195197
+           U68591  0.329854        0.014395
+
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname( 'mothur_datatypetest_true.mothur.axes' )
+        >>> Axes().sniff( fname )
+        True
+        >>> fname = get_test_fname( 'mothur_datatypetest_false.mothur.axes' )
+        >>> Axes().sniff( fname )
+        False
+        """
+        headers = get_headers(filename, sep='\t')
+        count = 0
+        col_cnt = None
+        all_integers = True
+        for line in headers:
+            if count == 0:
+                pass
+            elif col_cnt is None:
+                col_cnt = len(line)
+                if col_cnt < 2:
+                    return False
+            else:
+                if len(line) != col_cnt:
+                    return False
+                try:
+                    for i in range(1, col_cnt):
+                        check = float(line[i])
+                        # Check abs value is <= 1.0
+                        if abs(check) > 1.0:
+                            return False
+                        # Also test for whether value is an integer
+                        try:
+                            check = int(line[i])
+                        except ValueError:
+                            all_integers = False
+                except ValueError:
+                    return False
+            count += 1
+
+        if count > 0:
+            return not all_integers
+
+        return False
+
+
+class SffFlow(Tabular):
+    MetadataElement(name="flow_values", default="", no_value="", optional=True, desc="Total number of flow values", readonly=True)
+    MetadataElement(name="flow_order", default="TACG", no_value="TACG", desc="Total number of flow values", readonly=False)
+    file_ext = 'mothur.sff.flow'
+    """
+        http://www.mothur.org/wiki/Flow_file
+        The first line is the total number of flow values - 800 for Titanium data. For GS FLX it would be 400.
+        Following lines contain:
+        - SequenceName
+        - the number of useable flows as defined by 454's software
+        - the flow intensity for each base going in the order of TACG.
+        Example:
+          800
+          GQY1XT001CQL4K 85 1.04 0.00 1.00 0.02 0.03 1.02 0.05 ...
+          GQY1XT001CQIRF 84 1.02 0.06 0.98 0.06 0.09 1.05 0.07 ...
+          GQY1XT001CF5YW 88 1.02 0.02 1.01 0.04 0.06 1.02 0.03 ...
+    """
+    def __init__(self, **kwd):
+        Tabular.__init__(self, **kwd)
+
+    def set_meta(self, dataset, overwrite=True, skip=1, max_data_lines=None, **kwd):
+        Tabular.set_meta(self, dataset, overwrite, 1, max_data_lines)
+
+        headers = get_headers(dataset.file_name, sep='\t')
+        try:
+            flow_values = int(headers[0][0])
+            dataset.metadata.flow_values = flow_values
+        except:
+            pass
```

Comment:
I think this pass should at least give a warning to the log file? You can take a look at the following example: https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/datatypes/binary.py#L512